bodhi/server/models/models.py

```diff
@@ -1260,11 +1264,26 @@ def modify_bugs(self):
                     # https://github.com/fedora-infra/bodhi/issues/368#issuecomment-135155215
                     for bug in self.bugs:
                         if not bug.parent:
-                            log.debug("Closing tracker bug %d" % bug.bug_id)
-                            bug.close_bug(self)
+                            bad, good = self.get_bug_karma(bug)
+                            bug_karma = good + bad
+                            if bug_karma >= 0:
+                                log.debug("Closing tracker bug %d" % bug.bug_id)
+                                bug.close_bug(self)
+                            else:
+                                log.debug('Adding stable comment to bugs for %s', self.title)
+                                bug.add_comment(self)
                 else:
                     for bug in self.bugs:
-                        bug.close_bug(self)
+                        # Close only when the bug has good or no karma
+                        # https://github.com/fedora-infra/bodhi/issues/1160
+                        bad, good = self.get_bug_karma(bug)
+                        bug_karma = good + bad
+                        if bug_karma >= 0:
+                            log.debug("Closing tracker bug %d" % bug.bug_id)
+                            bug.close_bug(self)
+                        else:
+                            log.debug('Adding stable comment to bugs for %s', self.title)
+                            bug.add_comment(self)
```

Comment:
This code block seems to be identical to the one above (in the other half of the if statement) - let's make it a helper method so that the code isn't repeated (and so that we have guaranteed consistency in the behaviors if anyone changes one of them in the future.)